DBI.xs

```diff
@@ -1425,10 +1425,9 @@ dbih_setup_handle(pTHX_ SV *orv, char *imp_class, SV *parent, SV *imp_datasv)
     if (mg_find(SvRV(h), DBI_MAGIC) != NULL)
         croak(errmsg, neatsvpv(orv,0), imp_class, "already a DBI (or ~magic) handle");
 
-    strcpy(imp_mem_name, imp_class);
-    strcat(imp_mem_name, "_mem");
-    if ( (imp_mem_stash = gv_stashpv(imp_mem_name, FALSE)) == NULL)
-        croak(errmsg, neatsvpv(orv,0), imp_mem_name, "unknown _mem package");
+    imp_mem_name = sv_2mortal(newSVpvf("%s_mem", imp_class));
+    if ( (imp_mem_stash = gv_stashpv(SvPVbyte_nolen(imp_mem_name), FALSE)) == NULL)
```

Comment:
Why you are not using `gv_stashsv` which takes directly perl scalar? Then you do not have to extract PV buffer from scalar.

Reply:
Because I'm not verse in XS and I don't know the myriads of macros Perl provides. Fixed in 36f2a2c5fea36d7d47d6871e420286643460e71b commit.

Reply:
@ppisar I have also the myriads of macros Perl provides! :)